rezasm-source/rezasm-core/src/instructions/targets/input_target.rs

```diff
@@ -33,6 +36,38 @@ impl InputTarget {
     pub fn new_string(data: &String) -> InputTarget {
         Self::StringInput(data.clone())
     }
+
+    pub fn new_dereference_offset(
+        register: usize,
+        offset: i64,
+    ) -> Result<InputTarget, ParserError> {
+        if registry::is_valid_register_number(register) {
+            Ok(InputTarget::DereferenceInput(register, offset))
+        } else {
+            Err(ParserError::InvalidRegisterNumberError(register))
+        }
+    }
+
+    pub fn new_register(register: &usize) -> Result<InputTarget, ParserError> {
+        Ok(InputTarget::RegisterInput(register.clone()))
+    }
+
+    fn register_data(&self, simulator: &Simulator) -> Result<RawData, ParserError> {
+        let register = match self {
+            InputTarget::DereferenceInput(r, _) => {
+                simulator.get_registers().get_register_by_number(r.clone())
+            }
+            InputTarget::RegisterInput(r) => {
+                simulator.get_registers().get_register_by_number(r.clone())
+            }
+            _ => Err(ParserError::InternalError),
```

Comment:
Make InternalError an enum to be more descriptive. Also make ParserError::InternalError(InternalError) and implement Into<ParserError> for InternalError.

Reply:
Maybe make an enum type for various internal errors, then implement `Into` for cases like this where a different error type is needed.

Reply:
See issue #25